### PR TITLE
flakes: document nixConfig option

### LINKED ISSUE #5988: Document `nixConfig` attribute in `flake.nix`
**Describe the bug**

Nix flakes [support a `nixConfig` attribute](https://github.com/NixOS/nix/pull/4189) but this is not documented anywhere. The only references in the manual I can find at the moment are a brief mention in [nix develop](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-develop.html?highlight=nixConfig#description) and a mention in the [Nix 2.6 release notes](https://nixos.org/manual/nix/stable/release-notes/rl-2.6.html?highlight=nixConfig#release-26-2022-01-24), but neither reference explains what it is.

This should be documented in `nix help flake`.

**Nix version**

Tested locally with Nix 2.5.1 using `nix help flake`, and the manual links above were checked with Nix 2.6.0.

### PR DESCRIPTION
**Pull request by lincolnauster**
The docs here are pretty brief; tell me if more detail is warranted or there are any other issues.

Fixes #5988.

### CODE REVIEW

**Comment by edolstra** [comment]
Thanks. This should probably mention security, i.e. unless the option is whitelisted (currently "bash-prompt", "bash-prompt-suffix", "flake-registry") or `accept-flake-config` is set, the user will be prompted interactively whether they want to accept the setting.

**Comment by lincolnauster** [comment]
> This should probably mention security, i.e. unless the option is
> whitelisted (currently "bash-prompt", "bash-prompt-suffix",
> "flake-registry") or `accept-flake-config` is set, the user will be
> prompted interactively whether they want to accept the setting.

Done - definitely good to have that documented :)

File: src/nix/flake.md [hunk 1]
```diff
@@ -292,6 +292,12 @@ The following attributes are supported in `flake.nix`:
   value (e.g. `packages.x86_64-linux` must be an attribute set of
   derivations built for the `x86_64-linux` platform).
 
+* `nixConfig`: a set of `nix.conf` options to be set when evaluating any
+  part of a flake. In the interests of security, only a small set of
+  whitelisted options (currently `bash-prompt`, `brash-prompt-suffix`,
```
Comment (cole-h) [inline src/nix/flake.md:297]:
```suggestion
  whitelisted options (currently `bash-prompt`, `bash-prompt-suffix`,
```

**Review by edolstra — APPROVED** [review]

**Comment by lincolnauster** [comment]
Is there anything blocking merging here?

**Pull request merged**